hbase-diagnostics/src/main/java/org/apache/hadoop/hbase/PerformanceEvaluation.java

```diff
@@ -2853,310 +2919,61 @@ static TestOptions parseOpts(Queue<String> args) {
-      if (cmd.startsWith(valueRandom)) {
-        opts.valueRandom = true;
-        continue;
-      }
-
-      final String valueZipf = "--valueZipf";
-      if (cmd.startsWith(valueZipf)) {
-        opts.valueZipf = true;
-        continue;
-      }
-
-      final String period = "--period=";
-      if (cmd.startsWith(period)) {
-        opts.period = Integer.parseInt(cmd.substring(period.length()));
-        continue;
-      }
-
-      final String addColumns = "--addColumns=";
-      if (cmd.startsWith(addColumns)) {
-        opts.addColumns = Boolean.parseBoolean(cmd.substring(addColumns.length()));
-        continue;
-      }
-
-      final String inMemoryCompaction = "--inmemoryCompaction=";
-      if (cmd.startsWith(inMemoryCompaction)) {
-        opts.inMemoryCompaction =
-          MemoryCompactionPolicy.valueOf(cmd.substring(inMemoryCompaction.length()));
-        continue;
-      }
-
-      final String columns = "--columns=";
-      if (cmd.startsWith(columns)) {
-        opts.columns = Integer.parseInt(cmd.substring(columns.length()));
-        continue;
-      }
+      if (cmd.startsWith("--")) {
+        final String[] parts = cmd.substring(2).split("=", 2);
+        final String key = parts[0];
 
-      final String families = "--families=";
-      if (cmd.startsWith(families)) {
-        opts.families = Integer.parseInt(cmd.substring(families.length()));
-        continue;
-      }
-
-      final String caching = "--caching=";
-      if (cmd.startsWith(caching)) {
-        opts.caching = Integer.parseInt(cmd.substring(caching.length()));
-        continue;
-      }
-
-      final String asyncPrefetch = "--asyncPrefetch";
-      if (cmd.startsWith(asyncPrefetch)) {
-        opts.asyncPrefetch = true;
-        continue;
-      }
-
-      final String cacheBlocks = "--cacheBlocks=";
-      if (cmd.startsWith(cacheBlocks)) {
-        opts.cacheBlocks = Boolean.parseBoolean(cmd.substring(cacheBlocks.length()));
-        continue;
-      }
-
-      final String scanReadType = "--scanReadType=";
-      if (cmd.startsWith(scanReadType)) {
-        opts.scanReadType =
-          Scan.ReadType.valueOf(cmd.substring(scanReadType.length()).toUpperCase());
-        continue;
-      }
-
-      final String bufferSize = "--bufferSize=";
-      if (cmd.startsWith(bufferSize)) {
-        opts.bufferSize = Long.parseLong(cmd.substring(bufferSize.length()));
-        continue;
-      }
-
-      final String commandPropertiesFile = "--commandPropertiesFile=";
-      if (cmd.startsWith(commandPropertiesFile)) {
-        String fileName = String.valueOf(cmd.substring(commandPropertiesFile.length()));
-        Properties properties = new Properties();
         try {
-          properties
-            .load(PerformanceEvaluation.class.getClassLoader().getResourceAsStream(fileName));
-          opts.commandProperties = properties;
-        } catch (IOException e) {
-          LOG.error("Failed to load metricIds from properties file", e);
```

Comment:
I changed two things here.

* The original error message was a bit strange. Where did `metricIds` come from? I changed it to `Failed to load command properties from file`
* It used to print the error message and proceed. I decided to throw the exception instead.